src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -654,6 +654,8 @@ self =>
     def isRawBar   = isRawIdent && in.name == raw.BAR
     def isRawIdent = in.token == IDENTIFIER
 
+    def isIdentExceptRaw(except: Name) = isIdent && !(isRawIdent && in.name == except)
```

Comment:
I think we can simplify a bit as follows
```
// inline definitions
(in.token == IDENTIFIER || in.token == BACKQUOTED_IDENT) && !(in.token == IDENTIFIER && in.name == except)
// distribute negation
(in.token == IDENTIFIER || in.token == BACKQUOTED_IDENT) && (in.token != IDENTIFIER || in.name != except)
// commutativity of ||
(in.token == IDENTIFIER || in.token == BACKQUOTED_IDENT) && (in.name != except || in.token != IDENTIFIER)
// exclude middle
(in.token == IDENTIFIER || in.token == BACKQUOTED_IDENT) && (in.name != except || in.token == BACKQUOTED_IDENT)
// pull out `|| in.token == BACKQUOTED_IDENT`
(in.token == IDENTIFIER && in.name != except) || in.token == BACKQUOTED_IDENT
```
I guess we can reabstract now and get to `in.token == BACKQUOTED_IDENT || isIdentExcept(name)`.
 
Also, are there other cases where we should be allowing backquoted idents? (modulo #4935).